mantis-common/src/main/java/io/mantisrx/common/JsonSerializer.java

```diff
@@ -0,0 +1,80 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.mantisrx.common;
+
+import io.mantisrx.shaded.com.fasterxml.jackson.core.JsonProcessingException;
+import io.mantisrx.shaded.com.fasterxml.jackson.core.type.TypeReference;
+import io.mantisrx.shaded.com.fasterxml.jackson.databind.DeserializationFeature;
+import io.mantisrx.shaded.com.fasterxml.jackson.databind.ObjectMapper;
+import io.mantisrx.shaded.com.fasterxml.jackson.databind.SerializationFeature;
+import io.mantisrx.shaded.com.fasterxml.jackson.databind.ser.FilterProvider;
+import io.mantisrx.shaded.com.fasterxml.jackson.databind.ser.impl.SimpleFilterProvider;
+import io.mantisrx.shaded.com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+import java.io.IOException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JsonSerializer {
+    private static final Logger logger = LoggerFactory.getLogger(JsonSerializer.class);
+
+    private static final ObjectMapper defaultObjectMapper = new ObjectMapper()
+            .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+            .configure(SerializationFeature.FAIL_ON_EMPTY_BEANS, false)
+            .registerModule(new Jdk8Module());
+    public static final SimpleFilterProvider DEFAULT_FILTER_PROVIDER;
+
+    static {
+        DEFAULT_FILTER_PROVIDER = new SimpleFilterProvider();
+        DEFAULT_FILTER_PROVIDER.setFailOnUnknownId(false);
+    }
+
+    private static String toJSON(
+            ObjectMapper mapper,
+            FilterProvider filters,
+            Object object) throws JsonProcessingException {
+        if (filters == null) {
+            filters = DEFAULT_FILTER_PROVIDER;
+        }
+        return mapper.writer(filters).writeValueAsString(object);
+
+    }
+
+    public static <T> T fromJSON(
+            ObjectMapper mapper,
+            String json,
+            TypeReference<T> expectedType) throws IOException {
+        return mapper.readerFor(expectedType).readValue(json);
+    }
+
+    public static <T> T fromJSON(
+            ObjectMapper mapper,
+            String json,
+            Class<T> expectedType) throws IOException {
+        return mapper.readerFor(expectedType).readValue(json);
+    }
+
+    public <T> T fromJSON(String json, Class<T> expectedType) throws IOException {
+        return defaultObjectMapper.readerFor(expectedType).readValue(json);
+    }
+
+    public <T> T fromJSON(String json, TypeReference<T> expectedType) throws IOException {
+        return defaultObjectMapper.readerFor(expectedType).readValue(json);
+    }
+
+    public String toJson(Object object) throws IOException {
```

Comment:
nit: I'd call out the semantic difference between this and `static toJson` in comments. 
Also, could we make this static?

Reply:
I have removed all the static methods.